### PR TITLE
Add message ordering samples

### PR DESCRIPTION
**Pull request by jmdobry**
Also, moved client library instantiation back into the sample functions.

I went back on that after feedback received in my python and ruby pubsub-sample-style PRs.

### CODE REVIEW

File: pubsub/subscriptions.js [hunk 1]
```diff
@@ -168,8 +188,71 @@ function pullMessages (subscriptionName, callback) {
...
+
+  // Pulls messages. Set returnImmediately to false to block until messages are
+  // received.
+  subscription.pull({ returnImmediately: true }, (err, messages) => {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    messages.forEach((message) => {
```
Comment (ace-n) [inline pubsub/subscriptions.js]: Maybe include a comment that mentions that the (re-)ordering is being done by JS, and not by the API itself?

File: pubsub/system-test/subscriptions.test.js [hunk 2]
```diff
@@ -108,7 +108,46 @@ describe(`pubsub:subscriptions`, () => {
...
+        subscriptions.pullOrderedMessages(subscriptionNameOne, (err) => {
+          assert.ifError(err);
+          assert.equal(console.log.callCount, 0);
+          pubsub.topic(topicName).publish({ data: expected, attributes: { orderId: '1' } }, (err, secondMessageIds) => {
+            assert.ifError(err);
+            setTimeout(() => {
+              subscriptions.pullOrderedMessages(subscriptionNameOne, (err) => {
+                assert.ifError(err);
+                assert.equal(console.log.callCount, 1);
+                assert.deepEqual(console.log.firstCall.args, [`* %d %j %j`, secondMessageIds[0], expected, { orderId: '1' }]);
```
Comment (ace-n) [inline pubsub/system-test/subscriptions.test.js]: Is there any way to clean this up using e.g. `async.waterfall`?

File: pubsub/system-test/topics.test.js [hunk 3]
```diff
@@ -61,24 +61,52 @@ describe(`pubsub:topics`, () => {
...
+  });
+
+  it(`should publish ordered messages`, (done) => {
+    const topics = require('../topics');
+    pubsub.topic(topicName).subscribe(subscriptionName, { reuseExisting: true }, (err, subscription) => {
+      assert.ifError(err);
+      topics.publishOrderedMessage(topicName, message.data, () => {
+        setTimeout(() => {
+          subscription.pull((err, messages) => {
+            assert.ifError(err);
```
Comment (ace-n) [inline pubsub/system-test/topics.test.js]: `async.waterfall` these?

File: pubsub/topics.js [hunk 4]
```diff
@@ -105,8 +117,62 @@ function publishMessage (topicName, data, callback) {
...
+  };
+
+  topic.publish(message, (err, messageIds) => {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    // Update the counter value
+    setPublishCounterValue(+message.attributes.orderId + 1);
```
Comment (ace-n) [inline pubsub/topics.js]: Extra `+`, or something I don't yet know about ES2015?

Comment (jmdobry) [inline pubsub/topics.js]: That's coercion back to a number, apparently the attributes of a PubSub message must be a Map<string, string>

File: pubsub/system-test/topics.test.js (hunk 3 above)
Comment (jmdobry) [inline pubsub/system-test/topics.test.js]: Done.

File: pubsub/system-test/subscriptions.test.js (hunk 2 above)
Comment (jmdobry) [inline pubsub/system-test/subscriptions.test.js]: Done.

File: pubsub/subscriptions.js (hunk 1 above)
Comment (jmdobry) [inline pubsub/subscriptions.js]: Done.

**Comment by jmdobry** [comment]
@ace-n Looks good?

**Comment by ace-n** [comment]
LGTM.

**Pull request merged**